### PR TITLE
Private Assets

### PR DESCRIPTION
**Pull request by dmitry-yudakov**
Original Asset3d functionality is now split into base Asset3d (options, meta) and UserAsset3d (user_id, meta, is_private). UserAsset3d keeps pointer to its base Asset3d.

API is mostly allowed to work with UserAsset3d. 
- adding - Asset3d is created if missing or existing one is used
- Edit/remove operations are allowed only for user's own objects. 
- GET operation reads user assets taking all public and current user's private assets; Meta of the response is merged from asset's base metadata and userAsset metadata.

Rest of the code uses asset3d as it used to.

- [x] UserAsset3d for keeping userID, meta and is_private for asset3d
- [x] Allow several assets with same asset_3d_id but different user_id
- [x] API
- [x] Seed data

### CODE REVIEW

File: README.md
Comment (jor-rit) [inline README.md:1]: +42 bonus points for updating Readme! ;p

File: database/assets_3d/db.go [hunk 1]
```diff
@@ -82,22 +113,29 @@ func (db *DB) UpsertAssets(ctx context.Context, assets3d []*entry.Asset3d) error
...
 }
 
-func (db *DB) RemoveAssetsByIDs(ctx context.Context, asset3dIDs []umid.UMID) error {
-	if _, err := db.conn.Exec(ctx, removeAssetsByIDsQuery, asset3dIDs); err != nil {
+// func (db *DB) RemoveAssetsByIDs(ctx context.Context, asset3dIDs []umid.UMID) error {
+// 	if _, err := db.conn.Exec(ctx, removeAssetsByIDsQuery, asset3dIDs); err != nil {
+// 		return errors.WithMessage(err, "failed to exec db")
+// 	}
+// 	return nil
+// }
```
Comment (jor-rit) [inline database/assets_3d/db.go]: just delete the commented out func

File: universe/assets_3d/api.go [hunk 2]
```diff
@@ -18,14 +18,14 @@ func (a *Assets3d) RegisterAPI(r *gin.Engine) {
...
 				authorizedAdmin.POST("/upload", a.apiUploadAsset3d)
-				authorizedAdmin.DELETE("", a.apiRemoveAssets3dByIDs)
+				// authorizedAdmin.DELETE("", a.apiRemoveAssets3dByIDs)
 			}
 
 			assets3d.GET("", a.apiGetAssets3d)
-			assets3d.GET("/meta", a.apiGetAssets3dMeta)
-			assets3d.GET("/options", a.apiGetAssets3dOptions)
+			// assets3d.GET("/meta", a.apiGetAssets3dMeta)
+			// assets3d.GET("/options", a.apiGetAssets3dOptions)
```
Comment (jor-rit) [inline universe/assets_3d/api.go]: again, just delete 'em. We have git history if these ever need to come back :)

File: universe/assets_3d/assets_3d.go [hunk 3]
```diff
@@ -65,15 +68,67 @@ func (a *Assets3d) CreateAsset3d(asset3dID umid.UMID) (universe.Asset3d, error)
 	return asset3d, nil
 }
 
-func (a *Assets3d) FilterAssets3d(predicateFn universe.Assets3dFilterPredicateFn) map[umid.UMID]universe.Asset3d {
-	return a.assets.Filter(predicateFn)
+func (a *Assets3d) CreateAsset3dIfMissing(asset3dID umid.UMID) (universe.Asset3d, error, bool) {
```
Comment (jor-rit) [inline universe/assets_3d/assets_3d.go]: Not a blocker, but I'd say just `CreateAsset3d` here as the public interface name, document the return bool as 'created' and users should get the intent.

File: universe/assets_3d/api_assets_3d.go [hunk 4]
```diff
@@ -42,32 +44,62 @@ func (a *Assets3d) apiGetAssets3d(c *gin.Context) {
...
 		var category string
-		meta := asset3d.GetMeta()
 
+		if userAsset3d.IsPrivate() && userAsset3d.GetUserID() != userID {
+			return false
+		}
+
+		asset3d := userAsset3d.GetAsset3d()
+		if asset3d == nil {
+			fmt.Printf("asset3d is nil, strange! %+v\n", userAsset3d.GetAssetUserIDPair())
```
Comment (jor-rit) [inline universe/assets_3d/api_assets_3d.go]: log.Debugf

File: universe/assets_3d/api_assets_3d.go [hunk 5]
```diff
@@ -197,14 +187,28 @@ func (a *Assets3d) apiUploadAsset3d(c *gin.Context) {
 		return
 	}
 
+	fmt.Printf("Upload to mm response: %+v\n", response)
```
Comment (jor-rit) [inline universe/assets_3d/api_assets_3d.go]: delete, or log.Debugf if you really really want it (but just use debugger breakpoint I'd suggest :)

**Review by jor-rit — APPROVED** [review]

File: universe/assets_3d/api_assets_3d.go (hunk 5 above)
Comment (dmitry-yudakov) [reply universe/assets_3d/api_assets_3d.go]: Yeah, sorry about these - some leftovers :)

File: universe/assets_3d/assets_3d.go (hunk 3 above)
Comment (dmitry-yudakov) [reply universe/assets_3d/assets_3d.go]: There is a simpler and more standard CreateAsset3d, hence CreateAsset3dIfMissing :)
But yeah, one of them is enough, so removed the old one and renamed the new one to CreateAsset3d.

**Review by jor-rit — APPROVED** [review]

File: universe/user_asset_3d/user_asset_3d.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	a.entry.Meta = meta
+
+	return nil
+}
+
+func (a *UserAsset3d) IsPrivate() bool {
+	a.mu.Lock()
+	defer a.mu.Unlock()
```
Comment (jor-rit) [inline universe/user_asset_3d/user_asset_3d.go]: Should be a read lock, mu.RLock/RUnlock here

**Review by jor-rit — CHANGES_REQUESTED** [review]

**Review by jor-rit — APPROVED** [review]

**Pull request merged**